bin/persona-gmail-bridge.js

```diff
@@ -280,6 +267,25 @@ app.get('/authenticate/forward', validate({ email: 'gmail' }),
 
 app.get('/authenticate/verify', function (req, res) {
   // Check input precondition:
+  // The bridge should return the expected state token.
+  if (req.query.state !== req.session.state) {
```

Comment:
Also check to make sure one or the other is not falsey. If both are `undefined`, doesn't really help.